addons/hr_holidays/tests/test_allocations.py

```diff
@@ -385,3 +386,27 @@ def test_automatic_allocation_type(self):
         })
 
         self.assertEqual(allocation.allocation_type, 'regular')
+
+    def test_create_allocation_from_company_with_no_employee_for_current_user(self):
+        """
+            This test makes sure that the allocation can be created if the current company doesn't have an employee
+            linked to the loggedIn user, the form initializes with default employee the one linked to the user
+            now it initializes with empty employee
+        """
+        company = self.env['res.company'].create({
+            'name': 'Wick enterprises',
+        })
+        john_wick = self.env['hr.employee'].create({
+            'name': 'John Wick',
+            'company_id': company.id,
+        })
+        self.env.user.write({
+            'company_ids': [Command.set(company.ids)],
+            'company_id': company.id
+        })
+        allocation_form = Form(self.env['hr.leave.allocation'])
```

Comment:
```suggestion
self.user_hrmanager.employee_id = False
allocation_form = Form(self.env['hr.leave.allocation'].with_user(self.user_hrmanager))
```
It's faster to use already created records, we already have a test company and test user so we can use them